bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -413,13 +402,14 @@ public static boolean isGeneration2(String serviceName) {
     public static boolean isGeneration2(ThingTypeUID thingTypeUID) {
         String thingTypeID = thingTypeUID.getId();
         return thingTypeID.startsWith(THING_TYPE_SHELLYPLUS_PREFIX)
-                || thingTypeID.startsWith(THING_TYPE_SHELLYPRO_PREFIX) || thingTypeID.contains("mini")
-                || THING_TYPE_SHELLYPLUSWALLDISPLAY.equals(thingTypeUID)
-                || isBluSeries(thingTypeUID) | isBluSeries(thingTypeUID) || THING_TYPE_SHELLYBLUGW.equals(thingTypeUID);
+                || thingTypeID.startsWith(THING_TYPE_SHELLYPRO_PREFIX)
+                || thingTypeID.contains(THING_TYPE_SHELLYMINI_MIDDLE)
+                || GROUP_WALLDISPLAY_THING_TYPES.contains(thingTypeUID) || isBluSeries(thingTypeUID)
+                || THING_TYPE_SHELLYPLUSBLUGW.equals(thingTypeUID);
     }
 
     public static boolean isBluSeries(ThingTypeUID thingTypeUID) {
-        return BLU_SENSOR_THING_TYPES.contains(thingTypeUID) && !THING_TYPE_SHELLYBLUGW.equals(thingTypeUID);
+        return GROUP_BLU_THING_TYPES.contains(thingTypeUID) && !THING_TYPE_SHELLYPLUSBLUGW.equals(thingTypeUID);
```

Comment:
No longer needed, since excluded from group now:
```suggestion
        return GROUP_BLU_THING_TYPES.contains(thingTypeUID);
```

Reply:
correct, removed